api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -105,6 +107,16 @@ public CohortDefinition getAnyEncounterDuringPeriod() {
         return new MappedParametersCohortDefinition(cd, "onOrAfter", "startDate", "onOrBefore", "endDate");
     }
 
+    @DocumentedDefinition("codedObsSearchAdvanced")
+    public CohortDefinition getPatientsHavingObs() {
+        CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
```

Comment:
@andela-cnwankwo yes, we are recommending that you change the method name to match the library key.  Regarding the type of cohort definition to use, it depends on the concept chosen.  As @djazayeri  points out, it would be nice if we had a convenience CohortDefinition that could handle all types and delegate as needed, but this does not currently exist.  We could look to add this in the near term, but as of now, you would have to first let the user choose the "question" concept from the drop-down list, then - based on the data type of this concept - determine which end point to use.

If you look at the existing cohort builder, and you choose different concepts, you will get different "value" pickers based on the type of concept.  If a concept is a coded question, then you will get a a widget (I assume a drop-down or autocomplete) that allows you to select concepts as the answers to query for.